src/mosaic-node-manager.cc

```diff
@@ -125,65 +151,67 @@ namespace ns3 {
-            mobModel->SetPosition(position);
-            singleNode->AggregateObject(mobModel);
             
             // Associate the node with buildings for better radio propagation modeling
-            BuildingsHelper::Install (singleNode);
-            
+            BuildingsHelper::Install(singleNode);
+
             // Ensure that the mobility models of all nodes are consistent with their positions
             BuildingsHelper::MakeMobilityModelConsistent(); 
 
             // Install an LTE device on the node
-            NetDeviceContainer vehDev = m_lteHelper->InstallUeDevice(singleNode);
-
+            NetDeviceContainer ueDev = m_lteHelper->InstallUeDevice(singleNode);
+            m_ueDevs.Add(ueDev);
             // Install the internet stack on the node
             InternetStackHelper internet;
             internet.Install(singleNode);
 
             // Assign an IPv4 address to the LTE device
-            Ipv4InterfaceContainer vehicleIpIface = m_epcHelper->AssignUeIpv4Address(vehDev);
+            Ipv4InterfaceContainer vehicleIpIface = m_epcHelper->AssignUeIpv4Address(ueDev);
 
             // Set up static routing for the node to use the default gateway provided by the EPC helper
             Ipv4StaticRoutingHelper Ipv4RoutingHelper;
-            Ptr<Ipv4StaticRouting> vehicleStaticRouting = Ipv4RoutingHelper.GetStaticRouting(veh->GetObject<Ipv4>());
+            Ptr<Ipv4StaticRouting> vehicleStaticRouting = Ipv4RoutingHelper.GetStaticRouting(singleNode->GetObject<Ipv4>());
             vehicleStaticRouting->SetDefaultRoute(m_epcHelper->GetUeDefaultGatewayAddress(), 1);
 
             // Attach the LTE device to the eNodeB (base station)
-            m_lteHelper->Attach(vehDev);
+            m_lteHelper->Attach(ueDev);
 
             // Create and activate a sidelink bearer for V2X communication
-            Ptr<LteSlTft> tft = Create<LteSlTft>(LteSlTft::TRANSMIT, m_clientRespondersAddress, m_groupL2Address); 
-            m_lteV2xHelper->ActivateSidelinkBearer(Simulator::Now(), vehDev, tft);
-            
-            // Install the V2X sidelink configuration on the LTE device
-            m_lteHelper->InstallSidelinkV2xConfiguration(vehDev, m_ueSidelinkConfiguration);
-            
+            Ptr<LteSlTft> tft = Create<LteSlTft>(LteSlTft::BIDIRECTIONAL, m_clientRespondersAddress, m_groupL2Address); 
+            m_lteV2xHelper->ActivateSidelinkBearer(Simulator::Now(), ueDev, tft);
+
             m_groupL2Address++;
+            m_clientRespondersAddress = Ipv4AddressGenerator::NextAddress (Ipv4Mask ("255.255.0.0"));
+
+            // Install the V2X sidelink configuration on the LTE device
+            m_lteHelper->InstallSidelinkV2xConfiguration(ueDev, m_ueSidelinkConfiguration);            
+
+            //Install mobility model
+            Ptr<ConstantVelocityMobilityModel> mobModel = CreateObject<ConstantVelocityMobilityModel>();
```

Comment:
Is this constant velocity model also used in the DSRC node creation? The motion of our nodes is controlled by how the vehicles in CARLA/SUMO move, so I'm not sure if that's the correct model. Is there a specific MosaicMobilityModel that was used before?

Reply:
Yes, constant velocity mobility model is what MOSAIC originally used in NS-3 federate for DSRC.